core/node/core.go

```diff
@@ -88,10 +89,31 @@ func Dag(bs blockservice.BlockService) format.DAGService {
 }
 
 // OnlineExchange creates new LibP2P backed block exchange (BitSwap)
-func OnlineExchange(provide bool) interface{} {
+func OnlineExchange(cfg *config.Config, provide bool) interface{} {
 	return func(mctx helpers.MetricsCtx, lc fx.Lifecycle, host host.Host, rt routing.Routing, bs blockstore.GCBlockstore) exchange.Interface {
 		bitswapNetwork := network.NewFromIpfsHost(host, rt)
-		exch := bitswap.New(helpers.LifecycleCtx(mctx, lc), bitswapNetwork, bs, bitswap.ProvideEnabled(provide))
+		opts := []bitswap.Option{bitswap.ProvideEnabled(provide)}
+		if cfg.Internal.Bitswap.EngineBlockstoreWorkerCount == 0 {
```

Comment:
If we're going to do it this way (i.e. check for the default value and then pass that into the option as opposed to relying on defaults within go-bitswap) it'll be cleaner if you calculate the values and then do

```
opts := []bitswap.Option{
    bitswap.ProvideEnabled(provide),
    bitswap.EngineBlockstoreWorkerCount(bsWorkerCount),
    bitswap.TaskWorkerCount(taskWorkerCount),
    ...
}
```